bwaMem.wdl

```diff
@@ -216,23 +243,98 @@ task slicer {
   
 }
 
+
+task extractUMIs {
+        input {
+            String umiList = "umiList"
+            String outputPrefix = "extractUMIs_output"
+            File fastq1
+            File? fastq2
+            String pattern1 = "(?P<umi_1>^[ACGT]{3}[ACG])(?P<discard_1>T)|(?P<umi_2>^[ACGT]{3})(?P<discard_2>T)"
+            String pattern2 = "(?P<umi_1>^[ACGT]{3}[ACG])(?P<discard_1>T)|(?P<umi_2>^[ACGT]{3})(?P<discard_2>T)"
+            String modules = "barcodex-rs/0.1.2 rust/1.45.1"
+            Int jobMemory = 24
+            Int timeout = 12
+        }
+
+        parameter_meta {
+            umiList: "Reference file with valid UMIs"
+            outputPrefix: "Specifies the start of the output files"
+            fastq1: "FASTQ file containing read 1"
+            fastq2: "FASTQ file containing read 2"
+            pattern1: "UMI RegEx pattern 1"
+            pattern2: "UMI RegEx pattern 2"
+            modules: "Required environment modules"
+            jobMemory: "Memory allocated for this job"
+            timeout: "Time in hours before task timeout"
+        }
+
+        command <<<
+
```

Comment:
`set -euo pipefail` would be good to add here

Reply:
done